api/src/pcapi/connectors/titelive.py

```diff
@@ -6,55 +7,71 @@
 from pcapi.core import logging as core_logging
 from pcapi.utils import requests
+from pcapi.utils.cache import get_from_cache
 
 
 logger = logging.getLogger(__name__)
 
 
-def get_jwt() -> str:
+def get_jwt_token() -> str:
+    TITELIVE_JWT_CACHE_KEY = "api:titelive_jwt:cache"
+    TITELIVE_JWT_CACHE_TIMEOUT = 5 * 60  # 60 seconds
     url = f"{settings.TITELIVE_EPAGINE_API_AUTH_URL}/login/{settings.TITELIVE_EPAGINE_API_USERNAME}/token"
-
     payload = {"password": settings.TITELIVE_EPAGINE_API_PASSWORD}
 
-    try:
-        response = requests.post(url, json=payload)
-    except (urllib3_exceptions.HTTPError, requests.exceptions.RequestException) as e:
-        core_logging.log_for_supervision(
-            logger,
-            logging.ERROR,
-            "Titelive get jwt: Network error",
-            extra={
-                "exception": e,
-                "alert": "Titelive error",
-                "error_type": "network",
-                "request_type": "get-jwt",
-            },
-        )
-        raise requests.ExternalAPIException(is_retryable=True) from e
-
-    if not response.ok:
-        if 400 <= response.status_code < 500:
+    def _get_jwt() -> str:
+        try:
+            response = requests.post(url, json=payload)
+        except (urllib3_exceptions.HTTPError, requests.exceptions.RequestException) as e:
             core_logging.log_for_supervision(
                 logger,
                 logging.ERROR,
-                "Titelive get jwt: External error: %s",
-                response.status_code,
+                "Titelive get jwt: Network error",
                 extra={
+                    "exception": e,
                     "alert": "Titelive error",
-                    "error_type": "http",
-                    "status_code": response.status_code,
+                    "error_type": "network",
                     "request_type": "get-jwt",
-                    "response_text": response.text,
                 },
             )
-            raise requests.ExternalAPIException(True, {"status_code": response.status_code})
+            raise requests.ExternalAPIException(is_retryable=True) from e
+
+        if not response.ok:
+            if 400 <= response.status_code < 500:
+                core_logging.log_for_supervision(
+                    logger,
+                    logging.ERROR,
+                    "Titelive get jwt: External error: %s",
+                    response.status_code,
+                    extra={
+                        "alert": "Titelive error",
+                        "error_type": "http",
+                        "status_code": response.status_code,
+                        "request_type": "get-jwt",
+                        "response_text": response.text,
+                    },
+                )
+                raise requests.ExternalAPIException(True, {"status_code": response.status_code})
+
+        json_string = json.dumps(response.json())
+        return json_string
```

Comment:
Sauf si on veut s'assurer qu'on récupère bien du JSON, on peut simplifier en renvoyant directement le contenu de la réponse :

```suggestion
        return response.text
```